packages/webpack-cli/lib/utils/Compiler.js

```diff
@@ -93,9 +96,6 @@ class Compiler {
             }
             return this.generateOutput(outputOptions, stats, statsErrors, processingMessageBuffer);
         }
-        if (!outputOptions.watch && stats.hasErrors()) {
-            process.exitCode = 2;
-        }
     }
```

Comment:
This block was never executed because we have a return statement earlier.

Reply:
We should refactor it and avoid using `return`